src/components/JourneyPatternEditor/index.tsx

```diff
@@ -16,14 +15,19 @@ import { VEHICLE_MODE } from '../../model/enums';
 import ServiceJourney from '../../model/ServiceJourney';
 import { createUuid } from '../../helpers/generators';
 import { getJourneyPatternWithSwappedStopPoints } from './helpers';
+import DeleteActionChip from '../DeleteActionChip';
+import CopyActionChip from '../CopyActionChip';
+import CopyDialog from './CopyDialog';
 
 type Props = {
   journeyPattern: JourneyPattern;
   onSave: (journeyPattern: JourneyPattern) => void;
   onDelete?: () => void;
+  onCopy: (jpName: string) => void;
   spoilPristine: boolean;
   flexibleLineType?: FlexibleLineType;
   transportMode?: VEHICLE_MODE;
+  journeyPatternsNames: string[];
```

Comment:
A slightly better design here would be to inject a validation method for jp names instead of passing all current names in.